libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -363,10 +362,27 @@ void ContractLevelChecker::checkBaseConstructorArguments(ContractDefinition cons
 	// check that we get arguments for all base constructors that need it.
 	// If not mark the contract as abstract (not fully implemented)
 	for (ContractDefinition const* contract: bases)
-		if (FunctionDefinition const* constructor = contract->constructor())
-			if (contract != &_contract && !constructor->parameters().empty())
-				if (!_contract.annotation().baseConstructorArguments.count(constructor))
-					_contract.annotation().unimplementedDeclarations->push_back(constructor);
```

Comment:
If you don't fill `unimplementedDeclarations`, the `fullyImplemented` property on the ast json will be true if only constructor arguments are missing. I guess this is fine, since it will lead to an error.

Reply:
I'd argue that missing constructor arguments does not make it not fully implemented. It's buggy but it does have all the functions.

If it does, we should treat other errors related to wrong base constructor arguments the same way. I.e too many args or wrong types or zero args given on the modifier list of the constructor. Currently all of these are handled in `TypeChecker` and do not affect these annotations.